### PR TITLE
fix: use proper yvUSDC vault address

### LINKED ISSUE #612: V2 (Alpha): USDC Yearn vault displaying incorrect APY
As a user, I want the yield displayed in the USDC yearn vault to display the same APR/APY as going to Yearn directly. 

Currently, our yearn vault shows a slightly different APR/APY.  However, all other assets are correct!

- Visit earn section of app.shapeshift...https://app.shapeshift.com/earn/staking-vaults
- Notice the APR/APY of USDC
- Visit Yearn directly: https://yearn.finance/#/vaults
- Notice the different APR for USDC

A/C:
The APR/APY of the USDC vault is the same on app.shapeshift as it is on the yearn site. 

yearn apr

yearn

### PR DESCRIPTION
**Pull request by jdambron**
## Description

Use propoer yvUSDC vault address: `0xa354F35829Ae975e850e23e9615b11Da1B3dC4DE`

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #612 

## Testing

1. Visit earn section of app.shapeshift...https://app.shapeshift.com/earn/staking-vaults
2. Notice the APY of USDC
3. Visit Yearn directly: https://yearn.finance/#/vaults
4. Notice the same APY for USDC

### CODE REVIEW

**Comment by 0xean** [comment]
Thanks for the PR @jdambron !

I looked this up and want to confirm with the engineers who worked on the integration.

Looking at the documents it looks like the v2 vault was correct in the original code (prior to your PR).  See here - https://docs.yearn.finance/getting-started/products/yvaults/vault-tokens#v2-yvault-tokens

So maybe the Yearn UI is showing the older v1 vault?

**Comment by jdambron** [comment]
@0xean you're welcome.

For sure. Let's check that with more knowledgeable engineers than me.

From what I got from the Yearn API it seemed the way to fix the difference in displayed APY, but I may be wrong.

Let me know if I can help.

**Review by theoboldfrazier — COMMENTED** [review]
We should consider about users that have already deposited into the old vault. I think better than replacing the old one, we should add a new one and differentiate with the symbol being yvUSDC 0.3.0 and yvUSDC 0.4.3

I think on the roadmap we have plans to add migration functionality where users in an old vault can migrate to the new one.

**Comment by reidrankin** [comment]
Verified 0xa354f35829ae975e850e23e9615b11da1b3dc4de is the latest USDC vault (according to `v2.registry.ychad.eth`)

**Comment by 0xean** [comment]
@reidrankin - thanks for confirming that, I am inclined to merge this then.

**Review by 0xean — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
> We should consider about users that have already deposited into the old vault. I think better than replacing the old one, we should add a new one and differentiate with the symbol being yvUSDC 0.3.0 and yvUSDC 0.4.3
> 
> I think on the roadmap we have plans to add migration functionality where users in an old vault can migrate to the new one.

yeah - this is going to hide balances on the yearn screen for users that have deposited into 0.3.0

@jdambron do you want to make the change that @theoboldfrazier is suggesting so that we can see both vaults, differentiated by their version number in the list?

File: src/features/earn/providers/yearn/constants/vaults.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const SUPPORTED_VAULTS: SupportedYearnVault[] = [
     type: EarnType.Vault
   },
   {
-    vaultAddress: toLower('0x5f18C75AbDAe578b483E5F43f12a39cF75b973a9'),
+    vaultAddress: toLower('0xa354f35829ae975e850e23e9615b11da1b3dc4de'),
```
Comment (0xdef1cafe) [inline src/features/earn/providers/yearn/constants/vaults.ts:39]: see comment, duplicate this vault, name the original `USDC 0.3.0` and the new one `USDC 0.4.3` until we have a migration pathway

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

**Comment by jdambron** [comment]
@0xdef1cafe sure. I will have a look at it tomorrow.

**Comment by jdambron** [comment]
@0xdef1cafe I did the modification to allow both USDC vaults at the same time.
I had to tweak a bit the existing code, as it was coded expecting only one vault for a specific token.

File: src/components/StakingVaults/StakingVaultRow.tsx [hunk 2]
```diff
@@ -115,7 +115,14 @@ export const StakingVaultRow = ({
           </SkeletonCircle>
         </Flex>
         <Skeleton isLoaded={isLoaded}>
-          <RawText size='lg' fontWeight='bold'>{`${asset.symbol} ${type}`}</RawText>
+          {/* TODO Remove USDC 0.3.0 test and special case when we have a migration path */}
```
Comment (0xdef1cafe) [inline src/components/StakingVaults/StakingVaultRow.tsx:118]: thanks for this @jdambron 

could you please implement this in a way such that we're not hard coding a check for the vault name here? e.g. if we have multiple vaults with the same symbol (yvUSDC) we do this conditional rendering, else do the normal rendering?

this just means when yearn releases 0.4.4 or 0.4.whatever we don't have special magic string conditions to update everywhere

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

**Comment by jdambron** [comment]
@0xdef1cafe I changed the logic to get the names from the `constant/vaults.ts` file.
It will be more flexible to manage changes on Yearn side (as it will be managed on ours).
I just had to rename the `USD` vault into `YCURVE` to keep the same values displayed.
No more special magic string!

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by 0xdef1cafe** [comment]
looks great, thanks for the changes!

**Pull request merged**